tests/test_lists.py

```diff
@@ -16,135 +18,194 @@ def create_list(self, *args, **kwargs):
-        self.assertEqual(l[-2], 2)
-        self.assertEqual(l[-3], 1)
-        self.assertRaises(IndexError, lambda: l[42])
-        self.assertRaises(IndexError, lambda: l[-42])
-
-        l = self.create_list([1, 2, 3, 2, 3])
-        self.assertEqual(l.index(2), 1)
-        self.assertEqual(l.index(2, 2), 3)
-        self.assertRaises(ValueError, l.index, 2, 2, 3)
-        self.assertEqual(l.count(2), 2)
+        for init in (self.create_list, list):
+            L = init([1, 2, 3])
+            self.assertEqual(L[0], 1)
+            self.assertEqual(L[1], 2)
+            self.assertEqual(L[2], 3)
+            self.assertEqual(L[-1], 3)
+            self.assertEqual(L[-2], 2)
+            self.assertEqual(L[-3], 1)
+            self.assertRaises(IndexError, lambda: L[42])
+            self.assertRaises(IndexError, lambda: L[-42])
+
+            L = init([1, 2, 3, 2, 3])
+            self.assertEqual(L.index(2), 1)
+            self.assertEqual(L.index(2, 2), 3)
+            self.assertRaises(ValueError, L.index, 2, 2, 3)
+            self.assertEqual(L.count(2), 2)
 
     def test_slice(self):
-        l = self.create_list([1, 2, 3])
-        self.assertEqual(list(l[0:1]), [1])
-        self.assertEqual(list(l[0:2]), [1, 2])
-        self.assertEqual(list(l[:]), [1, 2, 3])
-        self.assertEqual(list(l[0:-1]), [1, 2])
-        self.assertEqual(list(l[1:]), [2, 3])
-        self.assertEqual(list(l[1::1]), [2, 3])
-        self.assertEqual(list(l[1::2]), [2])
+        redis_list = self.create_list([0, 1, 2, 3])
+        python_list = [0, 1, 2, 3]
+
+        for index in [
+            slice(None, None),  # L[:]
+            slice(0, None, None),  # L[0:]
+            slice(1, None, None),  # L[1:]
+            slice(3, None, None),  # L[3:]
+            slice(4, None, None),  # L[4:]
+            slice(None, 0, None),  # L[:0]
+            slice(None, 1, None),  # L[:1]
+            slice(None, 3, None),  # L[:3]
+            slice(None, 4, None),  # L[:3]
+            slice(0, 0, None),  # L[0:0]
+            slice(1, 1, None),  # L[1:1]
+            slice(3, 3, None),  # L[3:3]
+            slice(3, 3, None),  # L[3:3]
+            slice(-1, None, None),  # L[-1:]
+            slice(-2, None, None),  # L[-2:]
+            slice(-4, None, None),  # L[-4:]
+            slice(None, -1, None),  # L[-1:]
+            slice(None, -2, None),  # L[-2:]
+            slice(None, -4, None),  # L[-4:]
+            slice(0, -1, None),  # L[0:-1]
+            slice(1, -1, None),  # L[1:-1]
+            slice(1, -2, None),  # L[1:-2]
+            slice(-3, -1, None),  # L[-3:-1]
+            slice(None, None, 1),  # L[::1]
+            slice(None, None, 2),  # L[::2]
+            slice(None, None, 3),  # L[::3]
+            slice(None, None, 4),  # L[::4]
+            slice(None, None, -1),  # L[::-1]
+            slice(None, None, -2),  # L[::-2]
+            slice(1, -1, 2),  # L[1:-1:1]
+            slice(1, -1, -2),  # L[1:-1:-2]
+        ]:
```

Comment:
aaah, if this was pytest, [`parametrize`](http://docs.pytest.org/en/latest/parametrize.html#parametrized-test-functions) would probably help here